File: Practica5/java/Modular-Estructurada/Estadisticas.java
import java.util.Scanner;

public class Estadisticas {
    public static float promedio(float[] numeros) {
    	float suma = 0;
        for (float num : numeros) {
            suma += num;
        }
        return suma / numeros.length;
    }
    public static float desviacion(float[] numeros, float promedio) {
        float sumaDiferencias = 0;
        for (float num : numeros) {
            sumaDiferencias += Math.pow(num-promedio, 2);
        }
        return (float)Math.sqrt(sumaDiferencias / (numeros.length-1));
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        float[] numeros = new float[10];

        System.out.println("Ingrese 10 números:");
        for (int i = 0; i < 10; i++) {
            numeros[i] = scanner.nextFloat();
        }

        float prom = promedio(numeros);
        float desv = desviacion(numeros, prom);

        System.out.printf("El promedio es %.2f%n", prom);
        System.out.printf("La desviación estándar es %.5f%n", desv);

        scanner.close();
    }
}

File: Practica5/Python/Modular-Estructurada/Estadisticas.py
import math

def obtener_datos():
    while True:
        try:
            numeros = list(map(float, input("Ingrese 10 números: ").split()))
            if len(numeros) != 10:
                raise ValueError("Debe ingresar exactamente 10 números.")
            return numeros
        except ValueError as e:
            print(f"Error: {e}. Intente nuevamente.")
def promedio(numeros):
    return sum(numeros) / len(numeros)
def desviacion(numeros):
    prom = promedio(numeros)
    suma_cuadrados = sum((x - prom) ** 2 for x in numeros)
    return math.sqrt(suma_cuadrados / (len(numeros) - 1))
def main():
    numeros = obtener_datos()
    prom = promedio(numeros)
    desv = desviacion(numeros)
    print(f"El promedio es {prom:.2f}")
    print(f"La desviación estándar es {desv:.5f}")

if __name__ == "__main__":
    main()
